### ISSUE TITLE
[feature] template

### ISSUE THREAD

**Issue by individual-it** [JankariTech/web-app-presentation-viewer#21]
Having a template system would be great, so that the whole presentation has a particular look.
Maybe even have a system where one can choose per slide from a pre-defined templates

**Comment by koebel** [comment]
Indeed, I guess there are two types of templates that could be useful for the user: 
- content templates (title slide, slide with few bullet points, slides with more text, slides with pictures, ...)
- design templates (definition of fonts, colours, etc.)

**Comment by koebel** [comment]
For slides where custom design templates are used, changing the OC theme from light to dark mode or vice versa shouldn't affect how the slides (background colour, font colour, etc. ), only UI elements of web. 
see also https://github.com/JankariTech/web-app-presentation-viewer/issues/17#issuecomment-1945474120

**Comment by koebel** [comment]
shall we split this into two separate issues? I personally think the content templates have higher priority than the design templates...

**Comment by saw-jan** [comment]
>shall we split this into two separate issues? I personally think the content templates have higher priority than the design templates... the later is kind of connected to [this issue](https://github.com/JankariTech/web-app-presentation-viewer/issues/10)

^ @grgprarup Could you help to decide on it?

**Comment by grgprarup** [comment]
> > shall we split this into two separate issues? I personally think the content templates have higher priority than the design templates... the later is kind of connected to [this issue](https://github.com/JankariTech/web-app-presentation-viewer/issues/10)
> 
> ^ @grgprarup Could you help to decide on it?

IMO there is no need for separation in this issue. This issue will focus on the templating system, and implement the different types of templates for slides as examples in content templates (https://github.com/JankariTech/web-app-presentation-viewer/issues/21#issuecomment-1945897864). For the latter, we have an open issue https://github.com/JankariTech/web-app-presentation-viewer/issues/10

**Comment by koebel** [comment]
ok, let's add the aspect of custom themes to https://github.com/JankariTech/web-app-presentation-viewer/issues/10 

I guess this is the core of this feature request is this:
 
```
As a user
I want to select from different content types of slides in my presentation
So that I can create a presentation where each slide suits the content I want to present
```

by types content types we're talking about things like 
- title slide
- subtitle slide
- slide with bullet point list
- slide with text and image
- two columns text slide
- image slide
- multi image slide
- ...

@individual-it @grgprarup please comment

**Comment by individual-it** [comment]
This project can be used for the templates (content types): https://github.com/JankariTech/revealjs-awesoMD
